### PR TITLE
use YAML syntax for ansible tasks

### PR DESCRIPTION
**Pull request by arbabnazar**
@edx/devops kindly review. Thanks

### CODE REVIEW

File: playbooks/roles/rabbitmq/tasks/main.yml [hunk 1]
```diff
@@ -3,149 +3,183 @@
...
+    src: "edx/app/rabbitmq/log-rabbitmq-queues.sh.j2"
+    dest: "{{ rabbitmq_app_dir }}/log-rabbitmq-queues.sh"
+    owner: "{{ rabbitmq_user }}"
+    group: "{{ rabbitmq_group }}"
+    mode: "0755"
 
-- name: set up a cron job to run the script
+- name: Set up a cron job to run the script
   cron:
     name: "log-queue-lenghts"
```
Comment (e0d) [inline playbooks/roles/rabbitmq/tasks/main.yml:57]: Please fix type while you are here.

File: playbooks/roles/rabbitmq/tasks/main.yml [hunk 2]
```diff
@@ -3,149 +3,183 @@
...
+    mode: "0644"
   register: cluster_configuration
 
-- name: install plugins
+- name: Install plugins
   rabbitmq_plugin:
-    names={{",".join(rabbitmq_plugins)}} state=enabled
+    names: "{{ item }}"
+    state: enabled
+  with_items: "{{ rabbitmq_plugins }}"
```
Comment (e0d) [inline playbooks/roles/rabbitmq/tasks/main.yml:108]: Please verify that this works as expected and also doesn't make the play much slower to run.

File: playbooks/roles/rabbitmq/tasks/main.yml [hunk 3]
```diff
@@ -3,149 +3,183 @@
...
+    path: "{{ rabbitmq_mnesia_folder }}"
+    state: absent
   when: erlang_cookie.changed or cluster_configuration.changed or rabbitmq_refresh
 
-- name: start rabbit nodes
-  service: name=rabbitmq-server state=restarted
+- name: Start rabbit nodes
+  service:
+    name: rabbitmq-server
+    state: restarted
```
Comment (e0d) [inline playbooks/roles/rabbitmq/tasks/main.yml]: Should this just be "started" as we stop above unconditionally?  I assume it throws a warning as it cannot stop rabbit.

File: playbooks/roles/rabbitmq/tasks/main.yml [hunk 4]
```diff
@@ -3,149 +3,183 @@
...
-    dest=/usr/local/bin/rabbitmqadmin
-
-- name: ensure rabbitmqadmin attributes
-  file: >
-    path=/usr/local/bin/rabbitmqadmin owner=root
-    group=root mode=0655
+- name: Install admin tools
+  get_url:
+    url: "http://localhost:{{ rabbitmq_management_port }}/cli/rabbitmqadmin"
+    dest: "/usr/local/bin/rabbitmqadmin"
```
Comment (e0d) [inline playbooks/roles/rabbitmq/tasks/main.yml:179]: Can we add a creates declaration here?

**Comment by e0d** [comment]
A few comments, but after I'm :+1: @edx/devops second review?

File: playbooks/roles/rabbitmq/tasks/main.yml (hunk 4 above)
Comment (arbabnazar) [inline playbooks/roles/rabbitmq/tasks/main.yml:179]: we don't need `creates` here becuase `get_url` is already idempotent

File: playbooks/roles/rabbitmq/tasks/main.yml (hunk 2 above)
Comment (arbabnazar) [inline playbooks/roles/rabbitmq/tasks/main.yml:108]: @e0d it doesn't make it slow, I have verified the time from the datadog plugin and no noticeable time difference

**Comment by e0d** [comment]
My comments have been addressed, would still like a second review on this one.  @edx/devops

File: playbooks/roles/rabbitmq/tasks/main.yml [hunk 5]
```diff
@@ -3,149 +3,184 @@
...
+    dest: "{{ rabbitmq_app_dir }}/log-rabbitmq-queues.sh"
+    owner: "{{ rabbitmq_user }}"
+    group: "{{ rabbitmq_group }}"
+    mode: "0755"
 
-- name: set up a cron job to run the script
+- name: Set up a cron job to run the script
   cron:
-    name: "log-queue-lenghts"
+    name: "log-queue-lengths"
```
Comment (jibsheet) [inline playbooks/roles/rabbitmq/tasks/main.yml]: Are we going to end up with 2 cron jobs on our existing rabbit servers if we rerun the play (which we do sometimes to update configuration)

**Comment by jibsheet** [comment]
My one question about the cron jobs, but everything else looks 👍

File: playbooks/roles/rabbitmq/tasks/main.yml (hunk 5 above)
Comment (arbabnazar) [inline playbooks/roles/rabbitmq/tasks/main.yml]: @e0d , @jibsheet raise a very good question, I think we need to leave it as misspelled now? or we can do it like this that check the wrong job name and delete it and add the new one. Please let me know about your thoughts. Thanks

**Pull request merged**